utils/definitions.ts

```diff
@@ -68,6 +70,12 @@ export interface IIndexInfoCommon {
 
 export type HashIndexInfo<Entity> = Index<HashCode<Entity>> & IIndexInfoCommon;
 
+export interface ILinkedInfoCommon {
+  controller: IRecordStateController<id[]>,
+}
+
+export type LinkedIndexInfo = ILinkedProperty & IIndexInfoCommon;
```

Comment:
This is way too hard to implement in first iteration
This feature would be useful for a very large states to access the StateController for constant time.

It should be millions of properties to impact this.